File: UI/UIEKrushiStdApp/src/app/Services/payment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '@environments/environment';
import { Payment } from '@models/Payment';
import { PaymentAddModel } from '@models/PaymentAddModel';
import { PaymentDetails } from '@models/paymentDetails';

@Injectable({
  providedIn: 'root',
})
export class PaymentService {
  private apiurl: string = environment.paymentsServiceUrl;

  constructor(private httpClient: HttpClient) {}

  getPaymentDetails(orderId: number): Observable<PaymentDetails> {
    let url = `${this.apiurl}/details/${orderId}`;
    return this.httpClient.get<PaymentDetails>(url);
  }

  getPayments(customerId: number): Observable<Payment[]> {
    let url = `${this.apiurl}/${customerId}`;
    return this.httpClient.get<Payment[]>(url);
  }

  addPayment(payment: PaymentAddModel): Observable<boolean> {
    let url = `${this.apiurl}`;
    return this.httpClient.post<boolean>(url, payment);
  }
}

File: Dotnet/NewAPI/PaymentsAPI/Services/PaymentService.cs
using Transflower.EKrushi.PaymentsAPI.Interfaces;
using Transflower.EKrushi.PaymentsAPI.Models;
using Transflower.EKrushi.PaymentsAPI.Repositories.Interfaces;

namespace Transflower.EKrushi.PaymentsAPI.PaymentDetailsRepository;

public class PaymentService : IPaymentService
{
    private readonly IPaymentRepository _repository;

    public PaymentService(IPaymentRepository repository)
    {
        _repository = repository;
    }

    public async Task<bool> AddPayment(PaymentAddModel payment)
    {
        return await _repository.AddPayment(payment);
    }

    public async Task<PaymentDetail> GetPaymentDetails(int orderId)
    {
        return await _repository.GetPaymentDetails(orderId);
    }

    public async Task<List<Payment>> GetPayments(int customerId)
    {
        return await _repository.GetPayments(customerId);
    }
}
